crates/provider-sdk/src/lib.rs

```diff
@@ -148,31 +150,125 @@ pub struct Context {
 /// The super trait containing all necessary traits for a provider
 pub trait Provider: ProviderHandler + WrpcNats + WrpcDispatch + Send + Sync + 'static {}
 
+/// Configuration of a link that is passed to a provider
+pub trait LinkConfig: Send + Sync {
+    /// Given that the link was established with the source as this provider,
+    /// get the target ID which should be a component
+    fn get_target_id(&self) -> &ComponentId;
+
+    /// Given that the link was established with the source as this provider,
+    /// get the target ID which should be a component
+    fn get_source_id(&self) -> &ComponentId;
+
+    /// Get the name of the link that was provided
+    fn get_link_name(&self) -> &ComponentId;
```

Comment:
This should just be a `String`, not a component ID

Reply:
*`LinkName`